modules/carbonRtdProvider.js

```diff
@@ -0,0 +1,218 @@
+  if (Array.isArray(carbonData?.profile?.audiences)) {
+    window.googletag.pubads().setTargeting('carbon_segment', carbonData.profile.audiences);
+  }
+
+  if (Array.isArray(carbonData?.context?.pageContext?.contextualclassifications)) {
+    var contextSegments = carbonData.context.pageContext.contextualclassifications.map(x => {
+      if (x.type && x.type == 'iab_intent' && x.id) {
+        return x.id;
+      }
+    }).filter(x => x != undefined);
+    window.googletag.pubads().setTargeting('cc-iab-class-id', contextSegments);
+  }
+
+  if (Array.isArray(carbonData?.context?.customTaxonomy)) {
+    var customTaxonomyResults = matchCustomTaxonomy(carbonData.context.customTaxonomy);
+    window.googletag.pubads().setTargeting('cc-custom-taxonomy', customTaxonomyResults);
+  }
+}
+
+export function setPrebidConfig(carbonData) {
+  const ortbData = { user: {}, site: {} };
+
+  if (Array.isArray(carbonData?.profile?.audiences)) {
+    var userSegments = carbonData.profile.audiences.map(function (x) { return { id: x } });
+    ortbData.user.data = [{
+      name: 'www.ccgateway.net',
+      ext: { segtax: 507 }, // 507 Magnite Custom Audiences
+      segment: userSegments
+    }];
+  }
+
+  if (Array.isArray(carbonData?.context?.pageContext?.contextualclassifications)) {
+    var contextSegments = carbonData.context.pageContext.contextualclassifications.map(function (x) {
+      if (x.type && x.type == 'iab_intent' && x.id) {
+        return { id: x.id };
+      }
+    }).filter(x => x !== undefined);
+    ortbData.site.content = ortbData.site.content || {};
+    ortbData.site.content.data = [{
+      name: 'www.ccgateway.net',
+      ext: { segtax: 2 },
+      segment: contextSegments
+    }];
+  }
+
+  if (Array.isArray(carbonData?.context?.customTaxonomy)) {
+    ortbData.site.ext = ortbData.site.ext || {};
+    ortbData.site.ext.data = ortbData.site.ext.data || {};
+    ortbData.site.ext.data.customTaxonomy = matchCustomTaxonomy(carbonData.context.customTaxonomy);
+  }
+
+  if (Array.isArray(carbonData?.context?.dealIds)) {
+    ortbData.site.ext = ortbData.site.ext || {};
+    ortbData.site.ext.data = ortbData.site.ext.data || {};
+    ortbData.site.ext.data.dealIds = carbonData.context.dealIds;
+  }
+
+  sourceConfig.mergeConfig({ortb2: ortbData});
+}
+
+export function updateRealTimeDataAsync(callback, moduleConfig, userConsent) {
+  let doc = window.top.document;
+  let pageUrl = `${doc.location.protocol}//${doc.location.host}${doc.location.pathname}`;
+  let parentId = moduleConfig.params.parentId;
+  let profileId = storage.getDataFromLocalStorage(PROFILE_ID_KEY) || '';
+
+  let reqUrl = new URL(`${DATA_HOST}/${MODULE_VERSION}/realtime/${parentId}`);
+  reqUrl.searchParams.append('profile_id', profileId);
+  reqUrl.searchParams.append('url', encodeURIComponent(pageUrl));
+
+  if (getGlobal().getUserIds && typeof getGlobal().getUserIds == 'function') {
+    let eids = getGlobal().getUserIdsAsEids();
```

Comment:
I think it makes more sense to check `getGlobal().getUserIdsAsEids()` instead of `getGlobal().getUserIds` since we are not using the latter